### PR TITLE
Delete metrics by key (fixes #18)

### LINKED ISSUE #18: Feature: Delete metrics by key
First of all: Great work! I have been using batsd for some months now and it works great for me.

I just wanted to clean up in my batsd database and remove old metrics that are not interesting to me anymore and noticed that that's actually harder than I thought: I would have to delete the corresponding keys in redis-cli (DEL command), and then figure out what data files belong to that metric. Therefore I would have to build MD5 hashes for the metric name and all rentention keys, check if each file exists, and if it does delete it. If the containing directory is empty then, I'd like to delete that too.

I think something like `./bin/batsd delete <key>` would be really nice to simplify things. Another way might be to extend the management interface on port 8126 by this 'delete' function. What do you guys think about that?

### PR DESCRIPTION
**Pull request by groe**
This pull request adds `bin/batsd delete <statistic>` to the command line interface.

When invoked it deletes the corresponding key and data point in redis and deletes all files of all retention levels associated with the given metric name. If this leaves any empty directories behind these will be deleted too.

The pull request also includes a test for the disk store deletion.

I appreciate your feedback!

### CODE REVIEW

File: bin/batsd [hunk 1]
```diff
@@ -69,6 +69,16 @@ begin
                  ms: Batsd::Handler::Timer.new(config)
                }
     Batsd::Receiver::Daemon.new(handlers, config).run
+  when :delete
+    statistic = ARGV[1]
+    if !statistic.nil? && !statistic.empty?
```
Comment (grk) [inline bin/batsd]: `if statistic && !statistic.empty?` ;)

File: bin/batsd [hunk 2]
```diff
@@ -69,6 +69,16 @@ begin
...
+      @deleter = Batsd::Deleter.new(config)
```
Comment (grk) [inline bin/batsd:74]: Do you need an instance variable here?

**Comment by grk** [comment]
:+1:

File: bin/batsd (hunk 1 above)
Comment (groe) [inline bin/batsd]: In this case your `if` is equivalent (and more easily readable), yep :)

File: bin/batsd (hunk 2 above)
Comment (groe) [inline bin/batsd:74]: Not necessarily, I just though why not? The delete command isn't something called very frequently I guess. However, it's not required of course.

**Comment by groe** [comment]
Thanks for your feedback, I made that more clear now.

**Comment by grk** [comment]
oh, that instance variable I was talking about was that you don't really need to use `@deleter` if you're not referencing it in other methods, simply `deleter` will be fine. The `statistic` local variable was fine. Sorry I didn't make myself clear enough.

**Comment by groe** [comment]
Oh, I see. Well, I made it an instance variable since `@truncator` is one too (is `@truncator` used anywhere else actually?)

**Comment by noahhl** [comment]
Merged! Thanks!

Made a couple of small changes. Most notably, you can now do `bin/batsd delete timers:mytimer` without specifying `mean|min|max|count|stddev|upper_90` and it will delete all of those (that's the common case I care about in practice; it's rare that I want to delete just one of those and not the others).

**Comment by groe** [comment]
Great! I haven't thought about deleting these timer-specific statistics - of course they shall be deleted by the delete command too. Perfect.

**Pull request merged**